src/compile.rs

```diff
@@ -101,11 +105,70 @@ fn compile_universal2(
     Ok(result)
 }
 
+/// We want to use `zig cc` as linker and c compiler. We want to call `python -m ziglang cc`, but
+/// cargo only accepts a path to an executable as linker, so we add a wrapper script. We then also
+/// use the wrapper script to pass arguments and substitute an unsupported argument.
+///
+/// We create different files for different args because otherwise cargo might skip recompiling even
+/// if the linker target changed
+fn prepare_zig_linker(context: &BuildContext) -> Result<PathBuf> {
+    let (zig_linker, target) = match context.platform_tag {
+        None | Some(PlatformTag::Linux) => (
+            "./zigcc-gnu.sh".to_string(),
+            "native-native-gnu".to_string(),
+        ),
+        Some(PlatformTag::Musllinux { x, y }) => (
+            format!("./zigcc-musl-{}-{}.sh", x, y),
+            format!("native-native-musl.{}.{}", x, y),
+        ),
+        Some(PlatformTag::Manylinux { x, y }) => (
+            format!("./zigcc-gnu-{}-{}.sh", x, y),
+            format!("native-native-gnu.{}.{}", x, y),
+        ),
+    };
+
+    let zig_linker_dir = dirs::cache_dir()
+        // If the really is no cache dir, cwd will also do
+        .unwrap_or_else(|| PathBuf::from("."))
+        .join(env!("CARGO_PKG_NAME"))
+        .join(env!("CARGO_PKG_VERSION"));
+    fs::create_dir_all(&zig_linker_dir)?;
+    let zig_linker = zig_linker_dir.join(zig_linker);
+
+    #[cfg(target_family = "unix")]
+    let mut custom_linker_file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .mode(0o700)
+        .open(&zig_linker)?;
+    #[cfg(not(target_family = "unix"))]
+    let mut custom_linker_file = File::create(&zig_linker)?;
+    // https://github.com/ziglang/zig/issues/10050#issuecomment-956204098
+    custom_linker_file.write_all(
+        format!(
+            r##"#!/bin/bash
+python -m ziglang cc ${{@/-lgcc_s/-lunwind}} -target {}
```

Comment:
BTW, I think most distros prefer using existing packages for dependencies, so can we also support plain zig installation which isn't from PyPI?

Reply:
Yes! Unfortunately we can't just use `zig` which would also allow using e.g. nightly zig because the zig-pypi deliberately doesn't support it (https://github.com/ziglang/zig-pypi/pull/4#issuecomment-994160817). We can add a feature flag to switch between zig-pypi and plain zig if distro people confirm that that works for them

Reply:
I think `zig cc` shouldn't have many break changes since it mostly acts like a clang compiler thus nighty zig should work fine? Anyway we should prefer to use the python ziglang package and check zig version (like >= 0.9.0) to ensure it's usable.